データベース設計

murakamiyu opened the pull request:
#WHAT
README.mdにデータベース設計を記述。

#WHY
データベース設計のため。

murakamiyu committed:
add_readme

MrUrakou commented on README.md:
```diff
@@ -1,32 +1,47 @@
-* Deployment instructions
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false|
```
nameで良いです。
そうしないとgroupのnameを表示させる場合にgroup.group_nameとなり逆に読みづらくなるので。

MrUrakou reviewed (commented):
アソシエーションを見直してください。
membersテーブルに
```
belongs_to :group
belongs_to :user
```
とあるので逆のgroupsとusersのテーブルからのmembersとのアソシエーションが必要です。
usersテーブルとgroupsテーブルは直接の多対多のアソシエーションが組めないので中間テーブルを介して互いのデータを取りに行きます。その書き方を調べてみましょう。

murakamiyu committed:
指摘箇所の変更

レビューありがとうございました。
以下２点の変更を行いましたので、確認よろしくお願い致します。
・groupsとusersテーブルにmembersとのアソシエーションを追加
・groupsテーブルのカラム名を変更

murakamiyu commented on README.md:
```diff
@@ -1,32 +1,47 @@
-* Database creation
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
```
レビューありがとうございます！
修正させていただきます。

murakamiyu committed:
指摘箇所修正

・外部キーのtypeをreference型に変更

yukiraicom commented on README.md:
```diff
@@ -1,35 +1,52 @@
-* Ruby version
+### Association
+- has_many : groups, through: :members
+- has_many : members
```
has_many : groups, through: :members
- has_many : members
この順番は逆の方がいいかと思います。いきなりthrough: :menbersとなってもわからないので👁

yukiraicom commented on README.md:
```diff
@@ -1,35 +1,52 @@
-* Deployment instructions
+### Association
+- has_many : users, through: :members
+- has_many : members
```
こちらも同様に逆が良いかと

yuki-ar reviewed (commented):
遅れてしまい大変申し訳ないです。

murakamiyu committed:
associationの順番変更
